src/post.ts

```diff
@@ -66,32 +66,38 @@ async function main() {
     const targetPath = await locateTarget(manifestPath);
     core.info(`Sweeping files from ${targetPath}.`);
 
+    const operationHandles: Promise<void>[] = [];
+
     // Iterate recursively over all files in `target`.
     for (const file of await fs.readdir(targetPath, { recursive: true })) {
         const filePath = path.join(targetPath, file);
-        const stat = await fs.stat(filePath);
-
-        // Skip over folders, since they cannot be deleted with `fs.rm()` and take up a minimal
-        // amount of space. Additionally, skip certain whitelisted files where it wouldn't make
-        // sense to delete them.
-        if (stat.isDirectory() || SKIPPED_FILES.includes(file)) {
-            core.debug(`Skipped ${filePath} because it is a directory or is whitelisted.`);
-            continue;
-        }
 
-        // If the file's last access time is older than the timestamp, delete it.
-        if (stat.atime.getTime() < stamp) {
-            if (core.isDebug()) {
-                core.info(`Deleting ${filePath} with \`atime\` of ${stat.atime}.`);
-            } else {
-                core.info(`Deleting ${filePath}.`);
-            }
-
-            await fs.rm(filePath);
-        } else {
-            core.debug(`Skipped ${filePath} because it was accessed after timestamp.`);
-        }
+        operationHandles.push(
+            fs.stat(filePath).then(async (stat) => {
+                // Skip over folders, since they cannot be deleted with `fs.rm()` and take up a minimal
+                // amount of space. Additionally, skip certain whitelisted files where it wouldn't make
+                // sense to delete them.
+                if (stat.isDirectory() || SKIPPED_FILES.includes(file)) {
+                    core.debug(`Skipped ${filePath} because it is a directory or is whitelisted.`);
+                    return;
+                }
+
+                // If the file's last access time is older than the timestamp, delete it.
+                if (stat.atime.getTime() < stamp) {
+                    if (core.isDebug()) {
+                        core.info(`Deleting ${filePath} with \`atime\` of ${stat.atime}.`);
+                    } else {
+                        core.info(`Deleting ${filePath}.`);
+                    }
+
+                    await fs.rm(filePath);
+                } else {
+                    core.debug(`Skipped ${filePath} because it was accessed after timestamp.`);
+                }
+            }));
     }
+
+    await Promise.all(operationHandles);
```

Comment:
Cool! I never realized you could do this! (Then again, my Javascript knowledge is sparse.)